blockstore.go

```diff
@@ -82,6 +84,14 @@ func gatewayError(err error) error {
 		return fmt.Errorf("%w: %s", gateway.ErrGatewayTimeout, err.Error())
 	}
 
+	// (Saturn) errors that support the RetryAfter interface need to be converted
+	// to the correct gateway error, such that the HTTP header is set.
+	for v := err; v != nil; v = errors.Unwrap(v) {
+		if r, ok := v.(interface{ RetryAfter() time.Duration }); ok {
+			return gateway.NewErrorRetryAfter(err, r.RetryAfter())
+		}
+	}
+
```

Comment:
Looks very hacky, but I love it. Now we support all errors that export the `RetryAfter() time.Duration` interface.